samcli/commands/local/cli_common/user_exceptions.py

```diff
@@ -5,12 +5,6 @@
 from samcli.commands.exceptions import UserException
 
 
-class InvokeContextException(UserException):
```

Comment:
Why replace instead of extending further? Wouldn't that make less changes overall and we can still `catch` `InvokeContextException` as we did before

Reply:
That was my original idea, but then I found these;
- `InvokeContextException` is not caught in any place, same goes with parent classes `UserException` and `ClickException`. 
- `InvokeContextException` is not used in any place that would make it convenient to leave there. After replacing the newer exceptions, no usage is left so I removed the entire class.

Reply:
But the hierarchy of exceptions can be no? For example, the main CLI would catch `InvokeContextException` and do something specific with it like communicating further help. If we flatten everything, we loose that. `InvokeContextException` is a user error Exception.